ui/test/nuxeo-lock-toggle-button.test.js

```diff
@@ -14,82 +14,318 @@ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 See the License for the specific language governing permissions and
 limitations under the License.
 */
-import { fixture, html, login, waitChanged } from '@nuxeo/nuxeo-elements/test/test-helpers.js';
+import { fixture, flush, html, login, waitForEvent } from '@nuxeo/nuxeo-elements/test/test-helpers.js';
 import { tap } from '@polymer/iron-test-helpers/mock-interactions.js';
+import { dom } from '@polymer/polymer/lib/legacy/polymer.dom';
 import '../actions/nuxeo-lock-toggle-button.js';
 /* eslint-disable no-unused-expressions */
 
 suite('<nuxeo-lock-toggle-button>', () => {
+  let button;
   let server;
+
+  const lockedDocument = {
+    'entity-type': 'document',
+    uid: '1',
+    contextParameters: {
+      permissions: ['Write'],
+    },
+    facets: [],
+    lockCreated: '2016-02-19T12:47:40.501+01:00',
+    lockOwner: 'Administrator',
+    type: 'File',
+  };
+
   setup(async () => {
     server = await login();
+    button = await fixture(
+      html`
+        <nuxeo-lock-toggle-button></nuxeo-lock-toggle-button>
+      `,
+    );
   });
 
-  suite('when a document is locked', () => {
-    let element;
-    setup(async () => {
-      const doc = {
-        'entity-type': 'document',
-        uid: '1',
-        lockCreated: '2016-02-19T12:47:40.501+01:00',
-        lockedBy: 'Administrator',
-        facets: [],
-        type: 'File',
+  suite('Button Visibility', () => {
+    test('Should not be visible when document is null', async () => {
+      button.document = null;
```

Comment:
not a huge fan of reusing the same fixture since a test can mess up the state for another one...
if it's annoying to have to recreate the fixture everytime we can have a helper method that takes a document and return the button, ie `makeBtn = (doc) => fixture(..)`
Hopefully we wouldn't need to flush right after a `fixture()`, if we do perhaps we should wrap it in our helpers to do so once and for all....

Reply:
It shouldn't since it is being created on `setup` method.
Of course we can have a helper method, but we should still make use of `setup` and the other unit test's framework capabilities!